flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -205,7 +216,7 @@ private StreamingJobGraphGenerator(
         this.defaultStreamGraphHasher = new StreamGraphHasherV2();
         this.legacyStreamGraphHashers = Arrays.asList(new StreamGraphUserHashHasher());
 
-        this.jobVertices = new HashMap<>();
+        this.jobVertices = new LinkedHashMap<>();
```

Comment:
Yes, it's needed, because we hope to set vertex/operator parallelisms/outputs in the same order as vertices' creation. I
I think the changes of order will not have any negative impacts ?